sdk/identity/azure-identity/src/managed_identity_source.cpp

```diff
@@ -118,6 +119,26 @@ void ValidateArcKeyFile(std::string const& fileName)
     throw AuthenticationException("Failed to get file size for '" + fileName + "'.");
   }
 }
+
+// Create IMDS-specific retry options that handle HTTP 410 responses with sufficient retry duration
+Azure::Core::Credentials::TokenCredentialOptions CreateImdsRetryOptions(
+    Azure::Core::Credentials::TokenCredentialOptions const& options)
+{
+  using Azure::Core::Http::HttpStatusCode;
+  
+  auto imdsOptions = options;
+  
+  // Add HTTP 410 (Gone) to the retryable status codes for IMDS
+  // According to Azure docs, IMDS returns 410 for the first 70 seconds when not ready
+  imdsOptions.Retry.StatusCodes.insert(HttpStatusCode::Gone);
+  
+  // Increase MaxRetries to ensure we can retry for at least 70 seconds when encountering 410
+  // With exponential backoff: 800ms + 1.6s + 3.2s + 6.4s + 12.8s + 25.6s + 51.2s = ~101s total
+  // This ensures we retry for longer than the required 70 seconds
+  imdsOptions.Retry.MaxRetries = 6;
```

Comment:
You are making all the attempts to have 6 retries, not just when we received 410.
To properly implement this, I believe there needs some support to be implemented in Azure Core.